src/dft/backends/mklcpu/commit.cpp

```diff
@@ -31,109 +31,133 @@
+                    }
+
+                    local_err = DftiCommitDescriptor(*handle_obj.get_pointer());
+                    (*status_obj.get_pointer()) = local_err;
+                });
+            })
+            .wait();
+
+        status = status_buffer.template get_access<sycl::access::mode::read>()[0];
         if (status != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
-                                         "DftiCreateDescriptor failed");
+            throw oneapi::mkl::exception("dft", "commit", "DftiCommitDescriptor failed");
         }
 
-        set_value(handle, config_values);
-
-        status = DftiCommitDescriptor(handle);
-        if (status != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
-                                         "DftiCommitDescriptor failed");
-        }
+        device_handle = handle_buffer.template get_access<sycl::access::mode::read>()[0];
     }
 
-    virtual void* get_handle() noexcept override {
-        return handle;
+    void* get_handle() noexcept override {
+        return reinterpret_cast<void*>(device_handle);
     }
 
     virtual ~commit_derived_impl() override {
-        DftiFreeDescriptor((DFTI_DESCRIPTOR_HANDLE*)&handle);
+        DftiFreeDescriptor((mklcpu_desc_t*)&device_handle);
     }
 
 private:
-    DFTI_DESCRIPTOR_HANDLE handle = nullptr;
-
-    constexpr DFTI_CONFIG_VALUE get_domain(domain d) {
-        if (d == domain::COMPLEX) {
-            return DFTI_COMPLEX;
-        }
-        else {
-            return DFTI_REAL;
-        }
-    }
-
-    constexpr DFTI_CONFIG_VALUE get_precision(precision p) {
-        if (p == precision::SINGLE) {
-            return DFTI_SINGLE;
-        }
-        else {
-            return DFTI_DOUBLE;
-        }
-    }
+    DFT_ERROR status;
+    mklcpu_desc_t device_handle = nullptr;
+    sycl::buffer<mklcpu_desc_t, 1> handle_buffer{ &device_handle, sycl::range<1>{ 1 } };
 
     template <typename... Args>
-    void set_value_item(DFTI_DESCRIPTOR_HANDLE hand, enum DFTI_CONFIG_PARAM name, Args... args) {
-        if (auto ret = DftiSetValue(hand, name, args...); ret != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception(
-                "dft/backends/mklcpu", "set_value_item",
-                "name: " + std::to_string(name) + " error: " + std::to_string(ret));
+    DFT_ERROR set_value_item(mklcpu_desc_t hand, enum DFTI_CONFIG_PARAM name, Args... args) {
+        DFT_ERROR value_err = DFTI_NOTSET;
+        value_err = DftiSetValue(hand, name, args...);
+        if (value_err != DFTI_NO_ERROR) {
+            throw oneapi::mkl::exception("dft", "set_value_item", std::to_string(name));
         }
+
+        return value_err;
     }
 
-    void set_value(DFTI_DESCRIPTOR_HANDLE& descHandle,
-                   const detail::dft_values<prec, dom>& config) {
+    void set_value(mklcpu_desc_t& descHandle, dft::detail::dft_values<prec, dom> config) {
```

Comment:
```suggestion
    void set_value(mklcpu_desc_t descHandle, const dft::detail::dft_values<prec, dom>& config) {
```
I believe `mklcpu_desc_t` is a pointer so I think it best to just pass by value.

Reply:
@FMarno can you recheck this ?